cache/in-memory/src/permission.rs

```diff
@@ -484,32 +489,37 @@ impl<'a> InMemoryCachePermissions<'a> {
         }
     }
 
+    /// Given a thread channel, retrieve its parent from the cache, and combine
+    /// parent and child permissions.
     fn parent_overwrites(
         &self,
-        channel_id: &Id<ChannelMarker>,
-        parent_overwrites: Option<Vec<PermissionOverwrite>>,
+        thread: &Channel,
     ) -> Result<Vec<PermissionOverwrite>, ChannelError> {
-        let channel = self.0.channels.get(channel_id).ok_or(ChannelError {
+        let parent_id = thread.parent_id.ok_or(ChannelError {
+            kind: ChannelErrorType::ParentChannelNotPresent {
+                thread_id: thread.id,
+            },
+            source: None,
+        })?;
+
+        let channel = self.0.channels.get(&parent_id).ok_or(ChannelError {
             kind: ChannelErrorType::ChannelUnavailable {
-                channel_id: *channel_id,
+                channel_id: parent_id,
             },
             source: None,
         })?;
 
         if channel.guild_id.is_some() {
-            if let Some(parent_overwrites) = parent_overwrites {
-                Ok([
-                    channel.permission_overwrites.clone().unwrap_or_default(),
-                    parent_overwrites.to_vec(),
-                ]
-                .concat())
-            } else {
-                Ok(channel.permission_overwrites.clone().unwrap_or_default())
-            }
+            let mut overwrites = Vec::new();
```

Comment:
Optimization wise, can we `unwrap_or(&[])` the two slices early and then create this `Vec` with the length of the two? Or would that not be worth it?